components/progress-bar/script.js

```diff
@@ -10,6 +10,24 @@ import template from './template.html';
 
 const BaseComponent = components.BaseComponent;
 
+/**
+ * Check if a value is a number and if not - log an error
+ * @param {string} propName - the name of the property that needs to be validated
+ * @param {any} value
+ * @returns {boolean} - true if it is a number or a string that can be cast to number, false if not
+ */
+function isAPositiveNumberValidation(propName, value) {
+    const parsed = parseInt(value);
+    const isValueNaN = isNaN(parsed);
+
+    if (isValueNaN || !isValueNaN && parsed < 0) {
+        console.error(`Unsupported type[${typeof parsed}] given for ${propName}. Possible values are positive numbers.`);
+        return false;
+    } else {
+        return true;
+    }
```

Comment:
```suggestion    
    if (isNaN(parsed)) { 
        console.error(`Unsupported type[${typeof parsed}] given for ${propName}. Possible values are positive numbers.`);
        return false;
     }
     
     if (parsed < 0) {
         console.error(`The passed value for ${propName} is not a positive number. Please use positive numbers only for ${propName}.`);
         return false;
     }
     
     return true;
```

Maybe like that so it handles and warns the users if a not positive value has been passed.